core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -224,9 +224,9 @@ class FsHistoryProviderSuite extends FunSuite with BeforeAndAfter with Matchers
       EventLoggingListener.initEventLog(new FileOutputStream(file))
     }
     val writer = new OutputStreamWriter(bstream, "UTF-8")
-    try {
+    Utils.tryWithSafeFinally {
       events.foreach(e => writer.write(compact(render(JsonProtocol.sparkEventToJson(e))) + "\n"))
-    } finally {
+    } {
```

Comment:
Not your fault, but this looks super ugly. I'd favor using something like the following instead of that `Utils` method:

    try { 
       // code
    } finally Utils.catchAndLogExceptions {
      // finally block that may throw
    }